### DISCUSSION TITLE
Is it possible to integrate Proxy API into your own server?

### DISCUSSION THREAD

**Discussion by ghettojezuz** [Unleash/unleash#8929] [Q&A]
Hi there!
I'm new to working with Unleash and feeling a bit confused about how to use it with a proxy.

I have an application that consists of the following parts:
1. A frontend application built with React.
2. A Node.js server (using Hapi.js) that acts as a proxy, forwarding requests from the frontend to backend APIs. This server is also responsible for logging, authentication, and other tasks.

I want to integrate the Unleash Proxy into my existing Node.js server instead of deploying a separate one.
For example, I’d like to write something like this:

```typescript
import { Server, Request, ResponseToolkit } from "@hapi/hapi";
import { createApp } from "@unleash/proxy";

const unleashApp = createApp({
    unleashUrl: "https://unleash-url.com/api",
    unleashApiToken: "server-key",
});

export const register = (server: Server) => {
    server.route({
        method: ["GET", "POST"],
        path: "/unleash/{path*}",
        handler: async (request: Request, h: ResponseToolkit) => {
            const result = await unleashApp(request.raw.req, request.raw.res);

            return h.response(result.response);
        },
    });
};

export const unleashProxyPlugin = {
    name: "unleashProxyPlugin",
    register,
};
```

Is it possible to set this up? Any advice or guidance would be greatly appreciated!

**Comment by gastonfournier** [comment]
Hi, we haven't tried that, we were using it with express framework, but it's already deprecated. You can use it if you want and check if you can use it. Our recommendation is to use unleash-edge or use the [Frontend API](https://docs.getunleash.io/reference/api/unleash/frontend-api) if you don't want to deploy a new server
